redis-cluster/base/statefulset.yaml

```diff
@@ -0,0 +1,211 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-scripts
+data:
+  prestop-redis.sh: |
+    #!/bin/bash
+    if [[ ! -z "$REDIS_PASSWORD" ]]; then
+      export REDISCLI_AUTH=$REDIS_PASSWORD
+    fi
+    
+    REDIS_ROLE=$(redis-cli role | head -1)
+    if [[ "$REDIS_ROLE" == "master" ]]; then
+      NODE_ID=$(redis-cli cluster myid)
+      FIRST_REPLICAS_INFO=($(redis-cli cluster replicas $NODE_ID | head -n 1))
+      FIRST_REPLICA_HOST="${FIRST_REPLICAS_INFO[1]%:*}"
+      redis-cli client pause 20000 ALL
+      redis-cli -h $FIRST_REPLICA_HOST cluster failover
+      
+      while [[ $(redis-cli role | head -1) != 'slave' ]]; do # wait until failover finished
+        sleep 1
+      done
+    fi
+  ping.sh: |
+    #!/bin/bash
+    if [[ ! -z "$REDIS_PASSWORD" ]]; then
+      export REDISCLI_AUTH=$REDIS_PASSWORD
+    fi
+    
+    response=$(timeout 5 redis-cli -h $1 -p $2 ping)
+    if [ "$response" != "PONG" ]; then
+      exit 1
+    fi
+    
+    check_output=$(redis-cli --cluster check localhost:$REDIS_PORT)
+    if [[ ! $check_output =~ "[OK] All 16384 slots covered" ]]; then
+      exit 1
+    fi
+
+  start-node.sh: |
+    #!/bin/bash 
+  
+    cp -r /etc/redis/mounted/config/. /etc/redis/config/
+    
+    echo "" >> /etc/redis/config/redis.conf
+    
+    if [[ ! -z "$REDIS_PASSWORD" ]]; then
+      echo "requirepass $REDIS_PASSWORD" >> /etc/redis/config/redis.conf
+      echo "masterauth $REDIS_PASSWORD" >> /etc/redis/config/redis.conf
+    fi
+    
+    echo "cluster-announce-ip $HOSTNAME.$REDIS_HEADLESS_SERVICE" >> /etc/redis/config/redis.conf  
+    
+    ARGS=("--include" "/etc/redis/config/redis.conf") #todo port
+    exec redis-server "${ARGS[@]}"
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: redis
+  labels:
+    app: redis
+  annotations:
+    "app.uw.systems/name": "Redis"
+    "app.uw.systems/description": "A key value store"
+    "app.uw.systems/repos.kubernetes-manifests": "https://github.com/utilitywarehouse/customer-support-manifests/databases/redis/cluster/base"
```

Comment:
I guess this needs an update as part of the move?

Reply:
Ah yes, will do good spot.